### PR TITLE
Move `AccountProvider` interface into `fp-em`

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

File: frame/evm/precompile/dispatch/src/mock.rs [hunk 1]
```diff
@@ -139,15 +139,17 @@ parameter_types! {
...
-	type AccountProvider = pallet_evm::NativeSystemAccountProvider<Self>;
+	type AccountProvider = fp_evm::NativeSystemAccountProvider<Self>;
 	type FeeCalculator = FixedGasPrice;
 	type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
 	type WeightPerGas = WeightPerGas;
 
 	type BlockHashMapping = pallet_evm::SubstrateBlockHashMapping<Self>;
-	type CallOrigin = EnsureAddressRoot<<Self::AccountProvider as pallet_evm::AccountProvider>::AccountId>;
+	type CallOrigin =
+		EnsureAddressRoot<<Self::AccountProvider as pallet_evm::AccountProvider>::AccountId>;
```
Comment (MOZGIII) [inline frame/evm/precompile/dispatch/src/mock.rs:149]:
```suggestion
	type CallOrigin = EnsureAddressRoot<<Self::AccountProvider as fp_evm::AccountProvider>::AccountId>;
```

File: frame/evm/precompile/dispatch/src/mock.rs [hunk 2]
```diff
@@ -139,15 +139,17 @@ parameter_types! {
...
 
-	type WithdrawOrigin = EnsureAddressNever<<Self::AccountProvider as pallet_evm::AccountProvider>::AccountId>;
+	type WithdrawOrigin =
+		EnsureAddressNever<<Self::AccountProvider as pallet_evm::AccountProvider>::AccountId>;
```
Comment (MOZGIII) [inline frame/evm/precompile/dispatch/src/mock.rs:152]:
```suggestion
	type WithdrawOrigin = EnsureAddressNever<<Self::AccountProvider as fp_evm::AccountProvider>::AccountId>;
```

File: frame/evm/src/lib.rs [hunk 3]
```diff
@@ -517,12 +518,14 @@ pub mod pallet {
 }
 
 /// Type alias for currency balance.
-pub type BalanceOf<T> =
-	<<T as Config>::Currency as Currency<<<T as Config>::AccountProvider as AccountProvider>::AccountId>>::Balance;
+pub type BalanceOf<T> = <<T as Config>::Currency as Currency<
+	<<T as Config>::AccountProvider as AccountProvider>::AccountId,
+>>::Balance;
```
Comment (MOZGIII) [inline frame/evm/src/lib.rs:523]: Why this change? A lot of diffs for seemingly no reason

File: primitives/evm/Cargo.toml [hunk 4]
```diff
@@ -16,6 +16,7 @@ scale-codec = { package = "parity-scale-codec", workspace = true }
 serde = { workspace = true, optional = true }
 # Substrate
 frame-support = { workspace = true }
+frame-system = { workspace = true }
```
Comment (MOZGIII) [inline primitives/evm/Cargo.toml:19]: No need to implement the `NativeSystemAccountProvider` at `fp_evm`, we can implement it back at  `pallet_evm` and save ourselves an import here.

**Review by MOZGIII — APPROVED** [review]
Can we revert the changes to mere formatting? They clutter the diff. If we need them, let's land them as a separate PR prior to this one, as I'm sure they can be applied automatically.

**Review by MOZGIII — CHANGES_REQUESTED** [review]
I meant to ask for changes above

File: frame/evm/precompile/dispatch/src/mock.rs (hunk 1 above)
Comment (dmitrylavrenov) [reply frame/evm/precompile/dispatch/src/mock.rs]: Fixed

File: frame/evm/precompile/dispatch/src/mock.rs (hunk 2 above)
Comment (dmitrylavrenov) [reply frame/evm/precompile/dispatch/src/mock.rs]: Fixed

File: frame/evm/src/lib.rs (hunk 3 above)
Comment (dmitrylavrenov) [reply frame/evm/src/lib.rs]: Fixed

File: primitives/evm/Cargo.toml (hunk 4 above)
Comment (dmitrylavrenov) [reply primitives/evm/Cargo.toml]: Fixed

**Comment by dmitrylavrenov** [comment]
> Can we revert the changes to mere formatting? They clutter the diff. If we need them, let's land them as a separate PR prior to this one, as I'm sure they can be applied automatically.

Formatter changes have been reverted.

**Review by MOZGIII — APPROVED** [review]
